WordPress/src/main/java/org/wordpress/android/ui/main/jetpack/staticposter/JetpackStaticPosterFragment.kt

```diff
@@ -47,6 +51,21 @@ class JetpackStaticPosterFragment : Fragment() {
         }
     }
 
+    @Composable
+    fun LockScreenOrientation(orientation: Int) {
+        DisposableEffect(orientation) {
+            requireActivity().requestedOrientation = orientation
+            onDispose {
+                // Although restore original orientation seems like the logical solution, it does not
+                // work in this case because dispose runs after the new fragment is created, which
+                // then just resets to the orientation that is started with. If we weren't using the
+                // same activity (like we are for tabs), this would work very nicely by setting orientation
+                // to user.
+                // no op
+            }
+        }
```

Comment:
**Informative:** Since you are not using the `onDispose` of this `DisposableEffect` you could use the simpler `SideEffect` wrapped inside a `remember` function to avoid the effect being executed every recomposition, similar to what `DisposableEffect` does.

Although I like your comment there saying why the `onDispose` is no-op and could save some time in the future when looking at it again, so I am marking this comment as "Informative" just to share some knowledge but no need to change it.

Reply:
Rookie compose mistake. Thanks for sharing your knowledge.